MigrationScript/Migration.ps1

```diff
@@ -55,81 +70,91 @@ function Main
         $providerType = Check-ProviderTypeInput
     }
 
-    #Comment the line below if not running on local PowerShell
-	$logger.LogInfo("Please select an account to connect to Azure Portal...")
+    $logger.LogInfo("Please select an account to connect to Azure Portal...")
     Connect-AzAccount
 
     $parsedv1ArmId = Get-ParsedArmId $amsv1ArmId
     $logger.LogInfoObject("Parsed AMSv1 ARM id - ", $parsedv1ArmId)
 
-    Set-CurrentContext $parsedv1ArmId.subscriptionId $tenantId
+    Set-CurrentContext $parsedv1ArmId.subscriptionId $tenantId -logger $logger;
 
+	$logger.LogInfo("Generating JWT token to access keyvault.");
     $KeyVaultToken = Get-Token("KeyVault")
 
     #Generate the KeyVault Name
+	$logger.LogInfo("Generating key vault name using ams v1 armid.");
     $keyVaultName = Get-KeyVaultName $amsv1ArmId
 
     # Add role assignment to read secrets from KeyVault
-    Add-KeyVaultRoleAssignment $keyVaultName
+    Add-KeyVaultRoleAssignment -keyVaultName $keyVaultName -logger $logger;
 
     # Fetch all the secrets from KeyVault
-    $listOfSecrets = Get-AllKeyVaultSecrets $KeyVaultToken $keyVaultName
+	$logger.LogInfo("Fetch all the secrets from Key Vault $keyVaultName.");
+    $listOfSecrets = Get-AllKeyVaultSecrets -KeyVaultToken $KeyVaultToken -keyVaultName $keyVaultName
 
     $saphanaTransformedList = New-Object System.Collections.ArrayList
     $sapNetWeaverTransformedList = New-Object System.Collections.ArrayList
     $unsupportedProviderList = New-Object System.Collections.ArrayList
 
+	if($providerType -notlike "saphana" -and $providerType -notlike "all" -and $providerType -notlike "sapnetweaver") {
```

Comment:
Ideally we have a list of providers , and we just check if the list contains the given ProviderType. List.contains("givenType"), not sure how much effort is this if too much change ignore